enterprise_access/apps/api/v1/tests/test_views.py

```diff
@@ -421,3 +423,234 @@ def test_create_happy_path(self):
+
+
+    def test_partial_update_happy_path(self):
+        """
+        Test that a customer configuration can be patched.
+        """
+
+        self.set_jwt_cookie(roles_and_contexts=[
+            {
+                'system_wide_role': SYSTEM_ENTERPRISE_ADMIN_ROLE,
+                'context': self.enterprise_customer_uuid_1
+            },
+        ])
+
+        customer_config = SubsidyRequestCustomerConfiguration.objects.get(
+            enterprise_customer_uuid=self.enterprise_customer_uuid_1
+        )
+
+        customer_config.subsidy_requests_enabled = False
+        customer_config.subsidy_type = None
+        customer_config.save()
+
+        payload = {
+            'subsidy_requests_enabled': True,
+            'subsidy_type': SubsidyTypeChoices.COUPON
+        }
+        response = self.client.patch(
+            CUSTOMER_CONFIGURATIONS_LIST_ENDPOINT + f'{self.enterprise_customer_uuid_1}/',
+            payload
+        )
+        assert response.status_code == status.HTTP_200_OK
+
+        customer_config = SubsidyRequestCustomerConfiguration.objects.get(
+            enterprise_customer_uuid=self.enterprise_customer_uuid_1
+        )
+
+        assert customer_config.subsidy_requests_enabled
+        assert customer_config.subsidy_type == SubsidyTypeChoices.COUPON
+
+
+    @mock.patch('enterprise_access.apps.api.tasks.delete_enterprise_subsidy_requests_task.delay')
+    @ddt.data(
+        (SubsidyTypeChoices.LICENSE, SubsidyTypeChoices.COUPON, SubsidyTypeChoices.LICENSE),
+        (SubsidyTypeChoices.COUPON, SubsidyTypeChoices.LICENSE, SubsidyTypeChoices.COUPON),
+        (SubsidyTypeChoices.LICENSE, None, SubsidyTypeChoices.LICENSE),
+        (SubsidyTypeChoices.COUPON, None, SubsidyTypeChoices.COUPON),
+        (None, SubsidyTypeChoices.LICENSE, None),
+    )
+    @ddt.unpack
+    def test_partial_update_deletes_old_requests(
+        self,
+        previous_subsidy_type,
+        new_subsidy_type,
+        expected_deleted_subsidy_type,
+        mock_delete_enterprise_subsidy_requests_task
+    ):
+        """
+        Test that old requests are deleted if the subsidy type changes.
+        """
+
+        self.set_jwt_cookie(roles_and_contexts=[
+            {
+                'system_wide_role': SYSTEM_ENTERPRISE_ADMIN_ROLE,
+                'context': self.enterprise_customer_uuid_1
+            },
+        ])
+
+        customer_config = SubsidyRequestCustomerConfiguration.objects.get(
+            enterprise_customer_uuid=self.enterprise_customer_uuid_1
+        )
+
+        customer_config.subsidy_requests_enabled = False
+        customer_config.subsidy_type = previous_subsidy_type
+        customer_config.save()
+
+        payload = {
+            'subsidy_requests_enabled': True,
+            'subsidy_type': new_subsidy_type
+        }
+        response = self.client.patch(
+            CUSTOMER_CONFIGURATIONS_LIST_ENDPOINT + f'{self.enterprise_customer_uuid_1}/',
```

Comment:
nit: Could `CUSTOMER_CONFIGURATIONS_LIST_ENDPOINT` go inside of the `f` string? :f: